admiral/pkg/clusters/envoyfilter.go

```diff
@@ -0,0 +1,153 @@
+		},
+		ObjectMeta: metaV1.ObjectMeta{
+			Name:      envoyFilterName,
+			Namespace: "istio-system",
+		},
+		Spec: *envoyfilterSpec,
+	}
+
+	admiralCache.RoutingPolicyFilterCache.Put(workloadIdentityKey+common.GetRoutingPolicyEnv(routingPolicy), rc.ClusterID, envoyFilterName)
+	var filter *networking.EnvoyFilter
+	//get the envoyfilter if it exists. If it exists, update it. Otherwise create it.
+	if eventType == admiral.Add || eventType == admiral.Update {
+		filter, err = rc.RoutingPolicyController.IstioClient.NetworkingV1alpha3().EnvoyFilters("istio-system").Get(envoyFilterName, metaV1.GetOptions{})
+		if err != nil {
+			log.Infof("msg=%s filtername=%s clustername=%s", "creating the envoy filter", envoyFilterName, rc.ClusterID)
+			filter, err = rc.RoutingPolicyController.IstioClient.NetworkingV1alpha3().EnvoyFilters("istio-system").Create(envoyfilter)
+			if err != nil {
+				log.Infof("Error creating filter: %v", filter)
+			}
+		} else {
+			log.Infof("msg=%s filtername=%s clustername=%s", "updating existing envoy filter", envoyFilterName, rc.ClusterID)
+			envoyfilter.ResourceVersion = filter.ResourceVersion
+			filter, err = rc.RoutingPolicyController.IstioClient.NetworkingV1alpha3().EnvoyFilters("istio-system").Update(envoyfilter)
+		}
+	}
+
+
+	return filter, err
+}
+
+func constructEnvoyFilterStruct(routingPolicy *v1.RoutingPolicy, workloadSelectorLabels map[string]string) *v1alpha3.EnvoyFilter {
+	var envoyFilterStringConfig string
+	for key, val := range routingPolicy.Spec.Config {
+		envoyFilterStringConfig += fmt.Sprintf("%s: %s\n", key, val)
+	}
+
+	configuration := types.Struct{
+		Fields: map[string]*types.Value{
+			"@type": {Kind: &types.Value_StringValue{StringValue: "type.googleapis.com/google.protobuf.StringValue"}},
+			"value": {Kind: &types.Value_StringValue{StringValue: envoyFilterStringConfig}},
+		},
+	}
+
+	vmConfig := types.Struct{
+		Fields: map[string]*types.Value{
+			"runtime": {Kind: &types.Value_StringValue{StringValue: "envoy.wasm.runtime.v8"}},
+			"code": {Kind: &types.Value_StructValue{StructValue: &types.Struct{Fields: map[string]*types.Value{
+				"local": {Kind: &types.Value_StructValue{StructValue: &types.Struct{Fields: map[string]*types.Value{
+					"filename": {Kind: &types.Value_StringValue{StringValue: "/etc/istio/extensions/dynamicrouter.wasm"}},
+				}}}},
+			}}}},
+		},
+	}
+
+	typedConfigValue := types.Struct{
+		Fields: map[string]*types.Value{
+			"config": {
+				Kind: &types.Value_StructValue{
+					StructValue: &types.Struct{
+						Fields: map[string]*types.Value{
+							"configuration": {Kind: &types.Value_StructValue{StructValue: &configuration}},
+							"vm_config":     {Kind: &types.Value_StructValue{StructValue: &vmConfig}},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	typedConfig := types.Struct{
+		Fields: map[string]*types.Value{
+			"@type":    {Kind: &types.Value_StringValue{StringValue: "type.googleapis.com/udpa.type.v1.TypedStruct"}},
```

Comment:
Please define string in the code as a constant

Reply:
It is common to use such values as string literals.
For example - https://github.com/solo-io/wasm/blob/72a5b0f2bd99957f0a412595943bc92a503c419e/tools/wasme/cli/pkg/deploy/filter/envoy_filter.go#L120-L128